libqrexec/exec.c

```diff
@@ -311,9 +311,15 @@ static int find_file(
     return rc;
 }
 
+bool exit_on_stdout_eof = false;
+bool exit_on_stdin_eof = false;
+
 static int load_service_config_raw(struct qrexec_parsed_command *cmd,
                                    char **user)
 {
+    static bool called_yet = false;
+    assert(!called_yet);
```

Comment:
This takes down the whole qrexec-agent service. openqa says everything is broken...

```
[2024-05-04 02:28:41] [   35.640158] systemd[1]: Started systemd-coredump@0-800-0.service - Process Core Dump (PID 800/UID 0).
[2024-05-04 02:28:41] [   35.643650] audit: type=1130 audit(1714804121.207:110): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-coredump@0-800-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[2024-05-04 02:28:41] [   35.996565] systemd-coredump[801]: [0;1;31mProcess 634 (qrexec-agent) of user 0 dumped core.
[2024-05-04 02:28:41] 
[2024-05-04 02:28:41] Module libeconf.so.0 from rpm libeconf-0.5.2-2.fc39.x86_64
[2024-05-04 02:28:41] Module libaudit.so.1 from rpm audit-3.1.2-8.fc39.x86_64
[2024-05-04 02:28:41] Module libpam.so.0 from rpm pam-1.5.3-3.fc39.x86_64
[2024-05-04 02:28:41] Module libvchan-xen.so.1 from rpm qubes-libvchan-xen-4.2.3-1.fc39.x86_64
[2024-05-04 02:28:41] Module libqrexec-utils.so.3 from rpm qubes-core-qrexec-4.2.18-1.2.fc39.x86_64
[2024-05-04 02:28:41] Module qrexec-agent from rpm qubes-core-qrexec-vm-4.2.18-1.2.fc39.x86_64
[2024-05-04 02:28:41] Stack trace of thread 634:
[2024-05-04 02:28:41] #0  0x00007402f0cbf834 __pthread_kill_implementation (libc.so.6 + 0x90834)
[2024-05-04 02:28:41] #1  0x00007402f0c6d8ee raise (libc.so.6 + 0x3e8ee)
[2024-05-04 02:28:41] #2  0x00007402f0c558ff abort (libc.so.6 + 0x268ff)
[2024-05-04 02:28:41] #3  0x00007402f0c5581b __assert_fail_base.cold (libc.so.6 + 0x2681b)
[2024-05-04 02:28:41] #4  0x00007402f0c65c57 __assert_fail (libc.so.6 + 0x36c57)
[2024-05-04 02:28:41] #5  0x00007402f0e2f664 load_service_config_raw (libqrexec-utils.so.3 + 0x6664)
[2024-05-04 02:28:41] #6  0x00007402f0e2f7bf load_service_config_v2 (libqrexec-utils.so.3 + 0x67bf)
[2024-05-04 02:28:41] #7  0x00005eadb2cf3293 main (qrexec-agent + 0x4293)
[2024-05-04 02:28:41] #8  0x00007402f0c5714a __libc_start_call_main (libc.so.6 + 0x2814a)
[2024-05-04 02:28:41] #9  0x00007402f0c5720b __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2820b)
[2024-05-04 02:28:41] #10 0x00005eadb2cf3d25 _start (qrexec-agent + 0x4d25)
[2024-05-04 02:28:41] ELF object binary architecture: AMD x86-64
[2024-05-04 02:28:41] [0m
[2024-05-04 02:28:41] [   36.000765] audit: type=1131 audit(1714804121.565:111): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-coredump@0-800-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[2024-05-04 02:28:41] [   36.002341] systemd[1]: systemd-coredump@0-800-0.service: Deactivated successfully.
[2024-05-04 02:28:41] [   36.024073] systemd[1]: [0;1;38;5;185mqubes-qrexec-agent.service: Main process exited, code=dumped, status=6/ABRT[0m
[2024-05-04 02:28:41] [   36.024239] systemd[1]: [0;1;38;5;185mqubes-qrexec-agent.service: Failed with result 'core-dump'.[0m
```

maybe set initial value to false here explicitly, instead of assuming the config is loaded only once per process?